src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectProtocolValidator.cs

```diff
@@ -429,28 +429,22 @@ public virtual HashAlgorithm GetHashAlgorithm(string algorithm)
             {
                 switch (algorithm)
                 {
-                    case SecurityAlgorithms.SHA256:
```

Comment:
I understand that. However you are making assumptions about what the algorithm will be used for.